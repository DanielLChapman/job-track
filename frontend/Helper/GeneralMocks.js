
import { CURRENT_USER_QUERY } from '../components/User';
import { fakeUser } from '../library/testUtils';

const fakeUserObj = fakeUser();
const notSignedInMocks = [
    {
      request: { query: CURRENT_USER_QUERY },
      result: { data: { authenticatedItem: null } },
    },
  ];
  

  const signedInMocks = [
    {
      request: { query: CURRENT_USER_QUERY },
      result: { data: { authenticatedItem: fakeUser() } },
    },
  ];

  export {
    notSignedInMocks,
    signedInMocks,
}