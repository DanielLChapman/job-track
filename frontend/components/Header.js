import React, { useState } from "react";
import PropTypes from "prop-types";
import CreateJob from "./Jobs/CreateJob";
import SignOut from "./Signout";
import SignIn from "./SignIn";
import SignUp from "./SignUp";
import Modal from "./Modal";
import { CSSTransition } from "react-transition-group";

Header.propTypes = {
    user: PropTypes.object,
};

function Header({ user }) {
    const [modal, setModal] = useState(false);
    const [modalCreateJob, setModalCreateJob] = useState(false);
    const [inUp, setInUp] = useState("In");

    return (
        <>
            <h1>Job Track</h1>
            {user && (
                <div className="header-button-class">
                    <button
                        type="button"
                        className="sign-in-button"
                        onClick={() => {
                            setModalCreateJob(!modalCreateJob);
                        }}
                    >
                        Create Job
                    </button>

                    <SignOut />
                    {modalCreateJob && (
                        <Modal closeFunc={setModalCreateJob}>
                            <CreateJob user={user} />
                        </Modal>
                    )}
                </div>
            )}
            {!user && (
                <div className="header-button-class">
                    <button
                        type="button"
                        className="sign-in-button"
                        onClick={() => {
                            setModal(!modal);
                        }}
                    >
                        Sign In / Up
                    </button>
                </div>
            )}
            {!user && (
                <CSSTransition
                    in={modal}
                    timeout={200}
                    classNames="modal-transition"
                    unmountOnExit
                    onExited={() => setModal(false)}
                >
                    
                    <Modal closeFunc={setModal} >
                        <div className="header">
                            {inUp === "In" && (
                                <button
                                    type="button"
                                    className="sign-in-up-selector-button"
                                    onClick={() => {
                                        setInUp("Up");
                                    }}
                                >
                                    Sign Up
                                </button>
                            )}
                            {inUp === "Up" && (
                                <button
                                    type="button"
                                    className="sign-in-up-selector-button"
                                    onClick={() => {
                                        setInUp("In");
                                    }}
                                >
                                    Sign In
                                </button>
                            )}
                        </div>
                        {inUp === "In" && (
                            <SignIn closeFunc={setModal} closeValue={!modal} />
                        )}
                        {inUp === "Up" && <SignUp />}
                    </Modal>
                </CSSTransition>
            )}
        </>
    );
}

export default Header;
