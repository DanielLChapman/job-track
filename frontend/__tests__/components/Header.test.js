import {act, screen, render} from '@testing-library/react';
import { MockedProvider } from '@apollo/react-testing';
import Header from '../../components/Header';
import {signedInMocks, notSignedInMocks} from '../../Helper/GeneralMocks'
import wait from 'waait';

describe('It insures the header contains the necessary information', () => {
    it ('has a title, and two buttons when not logged in', () => {
        const { container, debug} = render(
            <MockedProvider mocks={notSignedInMocks}>
                <Header />
            </MockedProvider>
        )

        expect(container).toMatchSnapshot();
    })

    it ('renders a different header when logged in', async () => {
        const {container: container2, debug: debug2} = render(
            <MockedProvider mocks={notSignedInMocks}>
                <Header />
            </MockedProvider>
        )
        const { container, debug} = render(
            <MockedProvider mocks={notSignedInMocks}>
                <Header />
            </MockedProvider>
        )

        await act(async() => {
            await wait(0);
        });
        expect(container).toMatchSnapshot();
        expect(container).not.toBe(container2);
    })
})