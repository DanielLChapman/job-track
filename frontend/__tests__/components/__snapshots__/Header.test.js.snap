// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It insures the header contains the necessary information has a title, and two buttons when not logged in 1`] = `
<div>
  <h1>
    Job Track
  </h1>
  <div
    class="header-button-class"
  >
    <button
      class="sign-in-button"
      type="button"
    >
      Sign In / Up
    </button>
  </div>
</div>
`;

exports[`It insures the header contains the necessary information renders a different header when logged in 1`] = `
<div>
  <h1>
    Job Track
  </h1>
  <div
    class="header-button-class"
  >
    <button
      class="sign-in-button"
      type="button"
    >
      Sign In / Up
    </button>
  </div>
</div>
`;
